source/shared/cpp/ObjectModel/SharedAdaptiveCard.cpp

```diff
@@ -84,14 +85,20 @@ std::shared_ptr<AdaptiveCard> AdaptiveCard::Deserialize(
     ParseUtil::ExpectTypeString(json, CardElementType::AdaptiveCard);
 
     std::string version = ParseUtil::GetString(json, AdaptiveCardSchemaKey::Version);
-    std::string minVersion = ParseUtil::GetString(json, AdaptiveCardSchemaKey::MinVersion);
     std::string fallbackText = ParseUtil::GetString(json, AdaptiveCardSchemaKey::FallbackText);
+    std::string language = ParseUtil::GetString(json, AdaptiveCardSchemaKey::Language);
+
+    double versionAsDouble = ::atof(version.c_str());
```

Comment:
>versionAsDouble [](start = 11, length = 15)

Is version required to be a float?

Reply:
As opposed to an int? I believe we need to support dot versions, so like, version 1.1.

---
In reply to: [172354938](https://github.com/Microsoft/AdaptiveCards/pull/1216#discussion_r172354938) [](ancestors = 172354938)

Reply:
no. Float makes sense. I meant that we read this as a string from the card payload but are we enforcing that it has to be a float representation? What happens for "version": "blah"?

---
In reply to: [172357010](https://github.com/Microsoft/AdaptiveCards/pull/1216#discussion_r172357010) [](ancestors = 172357010,172354938)

Reply:
Hmm... well atof will return 0, and renderer version will not be smaller than that, so we'll go along our merry way. I think that's probably reasonable, but I'll double check with Matt.

---
In reply to: [172358721](https://github.com/Microsoft/AdaptiveCards/pull/1216#discussion_r172358721) [](ancestors = 172358721,172357010,172354938)

Reply:
Matt says we should throw, will update

---
In reply to: [172362617](https://github.com/Microsoft/AdaptiveCards/pull/1216#discussion_r172362617) [](ancestors = 172362617,172358721,172357010,172354938)